oracle-linux-image-tools/cloud/vagrant-virtualbox/provision.sh

```diff
@@ -0,0 +1,196 @@
+  chmod 600 /home/vagrant/.ssh/authorized_keys
+  chown -R vagrant:vagrant /home/vagrant/.ssh
+
+  # Fix for issue #76, regular users can gain admin privileges via su
+  ex -s /etc/pam.d/su <<'EOF'
+/^account\s\+sufficient\s\+pam_succeed_if.so uid = 0 use_uid quiet$/
+:append
+# allow vagrant to use su, but prevent others from becoming root or vagrant
+account         [success=1 default=ignore] \\
+                                pam_succeed_if.so user = vagrant use_uid quiet
+account         required        pam_succeed_if.so user notin root:vagrant
+.
+:update
+:quit
+EOF
+
+  echo 'vag' > /etc/yum/vars/infra
+
+  # Configure grub to wait just 1 second before booting
+  sed -i 's/^GRUB_TIMEOUT=[0-9]\+$/GRUB_TIMEOUT=1/' /etc/default/grub
+  grub2-mkconfig -o /boot/grub2/grub.cfg
+
+  # Blacklist the floppy module to avoid probing timeouts
+  echo blacklist floppy > /etc/modprobe.d/nofloppy.conf
+  chcon -u system_u -r object_r -t modules_conf_t /etc/modprobe.d/nofloppy.conf
+
+  # Customize the initramfs
+  # Enable VMware PVSCSI support for VMware Fusion guests.
+  echo 'add_drivers+=" mptspi "' > /etc/dracut.conf.d/vmware-fusion-drivers.conf
+  restorecon /etc/dracut.conf.d/vmware-fusion-drivers.conf
+  # There's no floppy controller, but probing for it generates timeouts
+  echo 'omit_drivers+=" floppy "' > /etc/dracut.conf.d/nofloppy.conf
+  restorecon /etc/dracut.conf.d/nofloppy.conf
+  # Regenerate initrd
+  local current_kernel=$(uname -r)
+  ${DRACUT_CMD} -f "/boot/initramfs-${current_kernel}.img" "${current_kernel}"
+
+  # Set SELinux to enforcing
+  sed -i -e 's/^SELINUX\s*=.*/SELINUX=enforcing/' /etc/selinux/config
+
+  # Install additional Packages
+  yum install -y ${YUM_VERBOSE} wget \
+    oracle-epel-release-el7 \
+    oraclelinux-developer-release-el7 \
+    oracle-softwarecollection-release-el7
+
+  # Enable optional repos
+  yum-config-manager --enable  ol7_addons >/dev/null
+  yum-config-manager --enable  ol7_preview >/dev/null
+  yum-config-manager --enable  ol7_developer >/dev/null
+  yum-config-manager --enable  ol7_developer_EPEL >/dev/null
+  yum-config-manager --enable  ol7_optional_latest >/dev/null
+
+  # Add login banner
+  echo "
+Welcome to Oracle Linux Server release `cat /etc/os-release | grep ^VERSION= | grep -o "[0-9].[0-9]"` (GNU/Linux $(uname -r))
+
+The Oracle Linux End-User License Agreement can be viewed here:
+
+  * /usr/share/eula/eula.en_US
+
+For additional packages, updates, documentation and community help, see:
+
+  * http://yum.oracle.com/
+  " > /etc/motd
+
+}
+
+#######################################
+# Install Virtualbox guest agent
+# Globals:
+#   YUM_VERBOSE
+# Arguments:
+#   None
+# Returns:
+#   None
+#######################################
+cloud::install_agent()
+{
+  echo_message "Install guest agent"
+  local additions="/mnt/VBoxLinuxAdditions.run"
+  yum install -y ${YUM_VERBOSE} make gcc bzip2 kernel-uek-devel
```

Comment:
Is it worth installing `kernel-uek-devel-$(uname -r)` so that we get maching versions between the currently running kernel and devel package?

Reply:
It will be pulled automatically with `kernel-uek-devel` so it is not necessary.
(But there I realize there is a bug on that line, as the package name needs to be aligned on the selected kernel uek/rhck)

Reply:
Fixed in 6f8003a